### ISSUE TITLE
Page dpi prop is not affecting page height and width anymore

### ISSUE THREAD

**Issue by MichiLor** [diegomura/react-pdf#2885]
**Describe the bug**
By default, a A4 page with the default dpi of 72 resulted in a page size of 595px (width) times 842px (height). We used then a dpi setting of 146 in order to achieve a page size of 1207px times 1707px. This was a requirement due to design guidelines that are based on roughly this size.
This works quite well til this PR seems to have change how the dpi behaves: [fix: page dpi (#2771) · diegomura/react-pdf@8e6a832 (github.com)](https://github.com/diegomura/react-pdf/commit/8e6a8320f86354aff950c296a96bc41a33e9dab2)

**To Reproduce**
1. Define a A4 page with a dpi of 146 with the react-pdf version **3.4.4**:
```
<Page debug dpi={146} size="A4">
     <Text style={{ fontSize: 13 }}>This is a text</Text>
</Page>
```
2. Resulted in expected page size
![image](https://github.com/user-attachments/assets/0b32e6e2-4d0e-4001-96f3-863a3324f47d)

3. Bump version to **3.4.5**
4. Results now in a smaller pixel page size
![image](https://github.com/user-attachments/assets/40db41ea-3e3d-47d7-9955-a1b11fe24847)

Steps to reproduce the behavior including code snippet (if applies):

**Expected behavior**
An A4 page with a dpi of 146 should again result in a width of 1207px and a height of 1707px.

**Screenshots**
See above

**Desktop (please complete the following information):**

- OS: Windows
- Browser chrome
- React-pdf version v3.4.5

**Comment by glarivie** [comment]
Same here. Works fine without this code: https://github.com/diegomura/react-pdf/commit/8e6a8320f86354aff950c296a96bc41a33e9dab2

**Comment by nikischin** [comment]
Hi @MichiLor @glarivie! 

Thank you very much for your bug report!

Unfortunately the topic is a bit more complicated here than just reverting the change. This is unfortunately due to the spec of pdf, which always expects the unit to be `user points` which always expects points in an PDF to be` 1/72 of an inch`.

Please check out more details to this [here](https://stackoverflow.com/a/3457303/2804367) or [here](https://stackoverflow.com/a/50715270/2804367)

The thing with PDF is, you always want to have a font set to 12pt to be the same size no matter what dpi setting you have. You can check on Adobe Indesign and create two pdfs, one with 72dpi and one with 300dpi, and in both the font will have the same size (relative to the page size) if set to 12pt. This was not working before the fix and was therefore super confusing. Why should the complete layout of the page change when I change the dpi setting?

However, what you experiencing is the pixel value not representing the 300dpi equivalent of that pixel value. Since the pdf is vector anyhow, this is only an issue with the unit and not the resolution. However, I also came up with that fact and therefore implemented #2773 which should fix this issue. Could you please check out my PR and let me know if this does solve your issue?

**Comment by MichiLor** [comment]
While further analyzing our issue, I found out the same thing. Independent of the dpi, a font size of 12pt should always be displayed as the same size. We decided to just adapt our design specs, which solves my issue already. 
Thanks for the help!
